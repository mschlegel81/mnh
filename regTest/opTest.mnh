boolOp:=[' and ',' or ',' xor '];
realOp:=["+","*","-","/","&","^"];
intOp :=realOp|boolOp|[" mod "," div "];
comparator:=['>','<','=','==','!=','>=','<='];
testOperator(op,LHS,RHS)->print(LHS,op,RHS,' = ',expression("$x"&op&"$y")(LHS,RHS));
main->begin
  each(op,boolOp,testOperator(op,true,false));
  each(op,intOp ,testOperator(op,3   ,5));
  each(op,realOp,testOperator(op,3.0,5.0));
  each(op,boolOp,testOperator(op,[true,true,false,false],[true,false,true,false]));
  each(op,intOp ,testOperator(op,[1..10],[1,2,3,2,2,3,1,4,3,5]));
  each(op,comparator,testOperator(op,3,3.0));
  each(op,comparator,testOperator(op,3,[5,3,8]));
  each(op,comparator,testOperator(op,"'x'",7));
end;
