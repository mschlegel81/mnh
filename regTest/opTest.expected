true and false = false
true or false = true
true xor false = true
3+5 = 8
3*5 = 15
3-5 = -2
3/5 = 0.6
3&5 = 35
3^5 = 243
3 and 5 = 1
3 or 5 = 7
3 xor 5 = 6
3.0+5.0 = 8.0
3.0*5.0 = 15.0
3.0-5.0 = -2.0
3.0/5.0 = 0.6
3.0&5.0 = 3.05.0
3.0^5.0 = 243.0
[true,true,false,false] and [true,false,true,false] = [true,false,false,false]
[true,true,false,false] or [true,false,true,false] = [true,true,true,false]
[true,true,false,false] xor [true,false,true,false] = [false,true,true,false]
[1,2,3,4,5,6,7,8,9,10]+[1,2,3,2,2,3,1,4,3,5] = [2,4,6,6,7,9,8,12,12,15]
[1,2,3,4,5,6,7,8,9,10]*[1,2,3,2,2,3,1,4,3,5] = [1,4,9,8,10,18,7,32,27,50]
[1,2,3,4,5,6,7,8,9,10]-[1,2,3,2,2,3,1,4,3,5] = [0,0,0,2,3,3,6,4,6,5]
[1,2,3,4,5,6,7,8,9,10]/[1,2,3,2,2,3,1,4,3,5] = [1.0,1.0,1.0,2.0,2.5,2.0,7.0,2.0,3.0,2.0]
[1,2,3,4,5,6,7,8,9,10]&[1,2,3,2,2,3,1,4,3,5] = ["11","22","33","42","52","63","71","84","93","105"]
[1,2,3,4,5,6,7,8,9,10]^[1,2,3,2,2,3,1,4,3,5] = [1,4,27,16,25,216,7,4096,729,100000]
[1,2,3,4,5,6,7,8,9,10] and [1,2,3,2,2,3,1,4,3,5] = [1,2,3,0,0,2,1,0,1,0]
[1,2,3,4,5,6,7,8,9,10] or [1,2,3,2,2,3,1,4,3,5] = [1,2,3,6,7,7,7,12,11,15]
[1,2,3,4,5,6,7,8,9,10] xor [1,2,3,2,2,3,1,4,3,5] = [0,0,0,6,7,5,6,12,10,15]
